contracts/src/bridge/FastBridgeSender.sol

```diff
@@ -44,14 +44,30 @@ contract FastBridgeSender is IFastBridgeSender {
         bytes memory encodedData = abi.encode(_receiver, _calldata);
 
         emit OutboxMessage(_receiver, keccak256(encodedData), encodedData);
+    }
 
+    /**
+     * Sends an arbitrary message from one domain to another
+     * via the safe bridge mechanism, which relies on the chain's native bridge.
+     *
+     * It is unnecessary during normal operations but essential only in case of challenge.
+     *
+     * It may require some ETH (or whichever native token) to pay for the bridging cost,
+     * depending on the underlying safe bridge.
+     *
+     * @param _receiver The L1 contract address who will receive the calldata
+     * @param _calldata The receiving domain encoded message data.
+     */
+    function sendSafe(address _receiver, bytes memory _calldata) external payable {
         // The safe bridge sends the encoded data to the FastBridgeReceiver
         // in order for the FastBridgeReceiver to resolve any potential
         // challenges and then forwards the message to the actual
         // intended recipient encoded in `data`
         // TODO: For this encodedData needs to be wrapped into an
         // IFastBridgeReceiver function.
         // TODO: add access checks for this on the FastBridgeReceiver.
-        safebridge.sendCrossDomainMessage(address(fastBridgeReceiver), encodedData);
+        // TODO: how much ETH should be provided for bridging? add an ISafeBridge.bridgingCost()
```

Comment:
Bridging from L2 to L1 does not need any bridging cost/fees.

Reply:
Right. Given the existing native bridges, the fees would be required only for L1 to L2, and we are not using the SafeBridge for it right now. So it's more for future proofing.